### PR TITLE
Added string precision metadata

### PR DESCRIPTION
**Pull request by bstrzele**

### CODE REVIEW

File: src/capi_frontend/capi.cpp [hunk 1]
```diff
@@ -844,6 +844,13 @@ DLL_PUBLIC OVMS_Status* OVMS_Inference(OVMS_Server* serverPtr, OVMS_InferenceReq
 
     std::unique_ptr<ModelInstanceUnloadGuard> modelInstanceUnloadGuard;
     auto status = getModelInstance(server, req->getServableName(), req->getServableVersion(), modelInstance, modelInstanceUnloadGuard);
+    if (status.ok()) {
```
Comment (dkalinowski) [inline src/capi_frontend/capi.cpp:847]: Is it good place for validation? We already have validate step, check `RequestValidator<ovms::InferenceRequest, InferenceTensor, const InferenceTensor*, signed_shape_t>::validatePrecision`

File: src/serialization.cpp [hunk 2]
```diff
@@ -215,6 +217,8 @@ static void serializeContent(::inference::ModelInferResponse::InferOutputTensor&
         SERIALIZE_BY_DATATYPE(mutable_uint_contents, uint8_t)
     } else if (responseOutput.datatype() == "FP64") {
         SERIALIZE_BY_DATATYPE(mutable_fp64_contents, double)
+    } else if (responseOutput.datatype() == "STRING") {
+        responseOutput.mutable_contents()->add_bytes_contents((char*)tensor.data(), tensor.get_byte_size());
```
Comment (dkalinowski) [inline src/serialization.cpp:221]: @bstrzele  Why are you performing string serialization in this PR? It is out of scope in this release. What does tensor.data() return when the type is string?

**Review by dkalinowski — CHANGES_REQUESTED** [review]
Unit tests are missing

File: src/predict_request_validation_utils.cpp [hunk 3]
```diff
@@ -894,6 +895,9 @@ Status RequestValidator<ovms::InferenceRequest, InferenceTensor, const Inference
         SPDLOG_DEBUG("[servable name: {} version: {}] Invalid precision - {}", servableName, servableVersion, details);
         return Status(StatusCode::INVALID_PRECISION, details);
     }
+    if (tensor.getDataType() == OVMS_DATATYPE_STRING) {
+        return Status(StatusCode::INVALID_PRECISION, "Inference on models using string precision input with capi is unsupported");
```
Comment (dkalinowski) [inline src/predict_request_validation_utils.cpp:899]: Unit test needed

**Comment by atobiszei** [comment]
Merged already

**Comment by dkalinowski** [comment]
https://github.com/openvinotoolkit/model_server/pull/2341

**Pull request closed without merging**